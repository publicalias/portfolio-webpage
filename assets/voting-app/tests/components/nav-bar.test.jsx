"use strict";

//local imports

const NavBar = require("../../scripts/components/nav-bar");

const { testWrapper } = require("../test-helpers");

//global imports

const { newUser } = require("schemas/master");
const { reactTests } = require("test-helpers/react-tests");

//setup

beforeAll(reactTests.setup);

//nav bar

describe("nav bar", () => {

  const { testMount, testShallow } = testWrapper(NavBar);

  const testFilter = (id, data) => () => {

    const { props, wrapper } = testMount(data);

    const { actions: { menuSetFilter }, history } = props;

    wrapper.find(`.qa-nav-${id}`).simulate("click");

    expect(menuSetFilter.mock.calls).toEqual([
      [id, history]
    ]);

  };

  it("should match snapshot (default)", () => {

    const { wrapper } = testShallow();

    expect(wrapper).toMatchSnapshot();

  });

  it("should match snapshot (authenticated)", () => {

    const { wrapper } = testShallow({ user: newUser() });

    expect(wrapper).toMatchSnapshot();

  });

  it("should call menuSetFilter on click (all)", testFilter("all"));

  it("should call menuSetFilter on click (created)", testFilter("created", { user: newUser() }));

  it("should call menuSetFilter on click (voted)", testFilter("voted"));

  it("should call menuSetFilter on click (hidden)", testFilter("hidden"));

  it("should call menuOpenForm on click (form)", () => {

    const { props, wrapper } = testMount({ user: newUser() });

    const { actions: { menuOpenForm }, history } = props;

    wrapper.find(".qa-nav-form").simulate("click");

    expect(menuOpenForm.mock.calls).toEqual([
      [history]
    ]);

  });

});
