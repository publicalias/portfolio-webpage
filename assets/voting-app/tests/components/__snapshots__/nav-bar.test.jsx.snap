// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar should match snapshot (authenticated) 1`] = `
<div
  className="c-ui__nav-bar"
>
  <h3
    className="c-ui__nav-item qa-nav-all"
    key="0-313"
    onClick={[Function]}
  >
    All Polls
  </h3>
  <h3
    className="c-ui__nav-item qa-nav-created"
    key="0-728"
    onClick={[Function]}
  >
    My Polls
  </h3>
  <h3
    className="c-ui__nav-item qa-nav-voted"
    key="0-546"
    onClick={[Function]}
  >
    Voted
  </h3>
  <h3
    className="c-ui__nav-item qa-nav-hidden"
    key="0-620"
    onClick={[Function]}
  >
    Hidden
  </h3>
  <h3
    className="c-ui__nav-item qa-nav-form"
    key="0-436"
    onClick={[Function]}
  >
    New Poll
  </h3>
</div>
`;

exports[`nav bar should match snapshot (default) 1`] = `
<div
  className="c-ui__nav-bar"
>
  <h3
    className="c-ui__nav-item qa-nav-all"
    key="0-313"
    onClick={[Function]}
  >
    All Polls
  </h3>
  <h3
    className="c-ui__nav-item qa-nav-voted"
    key="0-546"
    onClick={[Function]}
  >
    Voted
  </h3>
  <h3
    className="c-ui__nav-item qa-nav-hidden"
    key="0-620"
    onClick={[Function]}
  >
    Hidden
  </h3>
</div>
`;
